ophyd/status.py

```diff
@@ -705,7 +708,7 @@ def __init__(self, device, **kwargs):
     def _handle_failure(self):
         super()._handle_failure()
         self.log.debug("Trying to stop %s", repr(self.device))
-        if hasattr(self.device, "stop"):
+        if hasattr(self.device, "stop") and self._call_stop_on_failure:
```

Comment:
micro-optimization:

```suggestion
        if self._call_stop_on_failure and hasattr(self.device, "stop"):
```

if we don't plan to use `stop`, don't bother doing the lookup.